src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassReader.java

```diff
@@ -2261,6 +2261,7 @@ public void run() {
                 currentClassFile = classFile;
                 List<Attribute.TypeCompound> newList = deproxyTypeCompoundList(proxies);
                 sym.setTypeAttributes(newList.prependList(sym.getRawTypeAttributes()));
+                TypeAnnotationMapper.addTypeAnnotationsToSymbol(sym, newList);
```

Comment:
I wonder if the new TypeAnnotationsMapper should actually live inside of ClassReader

Reply:
I have tentatively gone ahead and moved it into ClassReader